### PR TITLE
[HIBIKE] document usage of virtual devices

### PR DESCRIPTION
**Pull request by karthik-shanmugam**
- add a markdown file documenting the usage of virtual devices
- modify `spawn_virtual_devices.py`, moving the user editable section to the top.
- change the example list of virtual devices in `spawn_virtual_devices.py`
  - demonstrate all 3 supported device types
  - demonstrate support for spawning multiple devices of the same type
- minor bugfix: invoke `python3` instead of `python3.5` , since virtual devices don't require python3.5

### CODE REVIEW

**Comment by nikitakit** [comment]
LGTM, but I'm deferring to @yizow for review. Since the runtime project was previously unaware of how to use virtual devices, they should be the ones to reply whether this documentation changes that or not.

File: hibike/VIRTUAL_DEVICES.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+Socat essentially generates a pair of connected virtual serial ports. Anything written to one of these ports can be read from the other. This way, the virtual device program can connect to one end of the pair of serial ports, and a program that wishes to communicate with hibike devices, such as `hibike_process.py`, can connect to the other end.
+
+The call we make to spawn a pair of ports is socat is `socat -d -d pty,raw,echo=0 pty,raw,echo=0`.
+
+### virtual_device.py
+
+The python script virtual_device.py simulates a hibike_device.
+It takes in as command line arguments the type of device to emulate and a port to connect to.
+It has an external dependency on pyserial should be run with python3.
```
Comment (yizow) [inline hibike/VIRTUAL_DEVICES.md]: on pyserial *and* should should be run

File: hibike/VIRTUAL_DEVICES.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+If you wish to implement more virtual devices types, or change the behaviour of any of the virtual device types, this is the file to modify.
+
+### spawn_virtual_devices.py
+
+This is a helper script that runs multiple instances the previous two programs, socat and virtual_device.py, to generate multiple virtual hibike devices.
```
Comment (yizow) [inline hibike/VIRTUAL_DEVICES.md]: multiple instances *of* the previous

**Review by yizow — APPROVED** [review]
Just a few word typos. Marking as approved so you can just fix and merge.

**Pull request merged**